test/circuit.jl

```diff
@@ -226,3 +231,59 @@ end
         sigma_z(3), hadamard(4)])
     @test compare_circuits(new_circuit, expected_circuit)
 end
+
+@testset "printout" begin
+    
+    target=1
+
+    gate_list=[
+        sigma_x(target),
+        sigma_y(target),
+        sigma_z(target),
+        hadamard(target),
+        pi_8(target),
+        pi_8_dagger(target),
+        x_90(target),
+        x_minus_90(target),
+        y_90(target),
+        y_minus_90(target),
+        z_90(target),
+        z_minus_90(target),
+        rotation(target,pi,pi/2),
+        rotation_x(target,pi/3),
+        rotation_y(target,pi/4),
+        phase_shift(target,pi/7),
+        universal(target,pi/3,pi/12,pi/4)
+    ]
+
+    circuit=QuantumCircuit(qubit_count=2,gates=gate_list)
+
+    print(circuit)
+
+    connected_qubits=[2,1]
+
+    gate_list=[
+        ControlledGate( sigma_x,   connected_qubits),
+        ControlledGate( sigma_y,   connected_qubits),
+        ControlledGate( sigma_z,   connected_qubits),
+        ControlledGate( hadamard,  connected_qubits),
+        ControlledGate( pi_8,      connected_qubits),
+        ControlledGate( pi_8_dagger,   connected_qubits),
+        ControlledGate( x_90,      connected_qubits),
+        ControlledGate( x_minus_90,connected_qubits),
+        ControlledGate( y_90,      connected_qubits),
+        ControlledGate( y_minus_90,connected_qubits),
+        ControlledGate( z_90,      connected_qubits),
+        ControlledGate( z_minus_90,connected_qubits),
+        ControlledGate( rotation,      connected_qubits; params=[pi,pi/2]),
+        ControlledGate( rotation_x,    connected_qubits; params=[pi/3]),
+        ControlledGate( rotation_y,    connected_qubits; params=[pi/4]),
+        ControlledGate( phase_shift,   connected_qubits; params=[pi/7]),
+        ControlledGate( universal,     connected_qubits; params=[pi/3,pi/12,pi/4]),
+    ]
+
+    circuit=QuantumCircuit(qubit_count=2,gates=gate_list)
+
+    print(circuit)
```

Comment:
This isn't really a test. All that this does is make sure it doesn't crash. It doesn't test that what it does is correct. I would recommend either removing this test or extending this test to ensure that the output is what is expected.

Reply:
this was removed, each ControlledGate is tested individually for producing the correct labels